nix/run-profiling.nix

```diff
@@ -10,38 +12,38 @@ writeScriptBin "run-profiling" ''
   out_folder="''${out_tar%.*.*}"
   timeout="''${2:-4m}"
 
-  mkdir -p profile
-  mkdir -p $out_folder/raw
-  ${gnutar}/bin/tar -xzvf "$1" -C ./profile
-  cd profile
+  mkdir -p profile-tmp
+  mkdir -p $out_folder
+  ${gnutar}/bin/tar -xzvf "$1" -C ./profile-tmp
+  cd profile-tmp
 
   # dump version of kore-exec for posterity
-  ${kore-exec-prof}/bin/kore-exec --version > ../$out_folder/kore-exec.version
+  ${kore-exec}/bin/kore-exec --version > ../$out_folder/kore-exec.version
+
+  # generate .eventlog from non-profiled kore-exec to be used by kore-prof
+  timeout --foreground -s INT "$timeout" env GHCRTS='-l' PATH='$PATH:${kore-exec}/bin' ./kore-exec.sh || true
+  cp kore-exec.eventlog ../$out_folder/kore-prof.eventlog
```

Comment:
Why is this called `kore-prof`?

Reply:
it's an event log produced by an unprofiled version of kore-exec, so it's only useful for consumption by the kore-prof tool, thus the name.

Reply:
Should we also run `kore-prof` on it as well? I can't see a place where we do that in the current script.

Reply:
As @jberthold pointed out, we may want to filter out certain messages when the eventlog is big, so it is probably better to just add the eventlog and let the user run kore-prof on it? we can of course run kore-prof with default seetings but we will have double the data, as was the case when including raw eventlogs for the other profiles

Reply:
> produced by an unprofiled version of kore-exec, so it's only useful for consumption by the kore-prof tool
Small correction, it also contains the markers for state machine tracing (currently only unification is instrumented). But fine to name it like this.